.github/workflows/gem-build-and-install.yml

```
@@ -0,0 +1,95 @@
+name: Test gem build and installation
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  cancel-in-progress: true
+
+on:
+  pull_request:
+
+jobs:
+  native-package:
+    name: Compile native gem
+    runs-on: ubuntu-latest
+    strategy:
+      # fail-fast: true
+      matrix:
+        platform:
+          - x86_64-linux
+          # - arm-linux # needs https://github.com/matsadler/magnus/issues/17 ", "aarch64-linux", "x86_64-darwin", "arm64-darwin", "x64-mingw32", "x64-mingw-ucrt"
+          # - "aarch64-linux"
+
+          - "x86_64-darwin"
+          # - "arm64-darwin" # github actions does not support this
+
+          # - "x64-mingw-ucrt"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: oxidize-rb/actions/setup-ruby-and-rust@main
+        with:
+          ruby-version: "3.1"
+          rubygems: latest
+          bundler-cache: false
+          cargo-cache: true
+          cargo-vendor: true
+
+      - uses: oxidize-rb/cross-gem-action@main
+        with:
+          platform: ${{ matrix.platform }}
+          env: |
```

Should be good to support 2.7+

```suggestion
          ruby-versions: '3.1, 3.0, 2.7'
```

---

Yeah, but I don't wanna 😝 

The real reason is that I don't want to maintain two Windows architectures for the next few years. Ruby 3.1 [changed the toolchain string](https://rubyinstaller.org/2021/12/31/rubyinstaller-3.1.0-1-released.html) and I don't want to carry the cognitive load of "if < 3.1, do X." People will just have to deal. 🤷🏻‍♂️